server/src/proxy/grpc/write.rs

```diff
@@ -152,18 +195,128 @@ impl<Q: QueryExecutor + 'static> Proxy<Q> {
             .await?;
         }
 
-        let resp = WriteResponse {
+        Ok(WriteResponse {
             success: success as u32,
             header: Some(build_ok_header()),
             ..Default::default()
+        })
+    }
+
+    async fn maybe_forward_write(
+        &self,
+        req: WriteRequest,
+    ) -> Result<(WriteRequest, Vec<BoxFuture<Result<WriteResponse>>>)> {
+        // Split write request into multiple requests, each request contains table
+        // belong to one remote engine.
+        let tables = req
+            .table_requests
+            .iter()
+            .map(|table_request| table_request.table.clone())
+            .collect();
+
+        let route_data = self
+            .router
+            .route(RouteRequest {
+                context: req.context.clone(),
+                tables,
```

Comment:
Maybe we should let the route method be able to handle an iterator on tables in order to allocate a vector of tables here. How about leaving a todo here?